### PR TITLE
feat: Update directory app with services

### PR DESCRIPTION
**Pull request by connoratrug**
### What are the main changes you did

Update the app to allow user to request services 

functional:
- allow user to filter biobanks by service type
- allow add services to the request/cart
- add selected services to negotiator request
- show service details in service report 

non-functional:
- switch touched files to typescript
- fix issues found by typing
- switch touched files to script setup
- pullout ui components from 'smart' components
- use ui components and containers to have consistent style 
- remove unused code 

### How to test
- explain here what to do to test this (or point to unit tests)

### Checklist
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

**Comment by esthervanenckevort** [comment]
Looking at the user interface changes I think this is 95% what we would like to see. The card with tabs for collections, services and organisation details is quite clean. Also good that you made the split between collections and services in the request cart.

Some improvements:
- The services search facet *must* use the label instead of the name attribute to show the items.
- The service types is  quite a long list and the display *could* be improved by using the category to group the services in a tree. In the data model they all have a category, the category (Service Category ontology type) should be used as the top level of the tree.
- The data that is passed to the Negotiator does not correctly fill in the 'humanReadable', it prepends the actions to the text "User actions taken: " It *should* append the text instead. See:
```
{
"url":"https://preview-emx2-pr-4501.dev.molgenis.org",
"humanReadable":"Biobank services: sample-collection, sample-data-management, sample-processing, sample-quality-control, sample-storage. User actions taken: ",
"resources":[{"id":"bbmri-eric:serviceID:DE_1234","name":"Biobank Service"}]}
```

**Review by esthervanenckevort — APPROVED** [review]
Tested functionality:
- Filtering works & services are shown on the main page and can be added to the cart
Few omissions:
- Services are not shown on the biobank details page
- Services details page does not have an add button
- Service types in main page are not shown in hierarchy

File: apps/directory/src/components/biobankcards-components/BiobankCard.vue [hunk 1]
```diff
@@ -1,313 +1,190 @@
...
+    ) ?? []
+  );
+});
+
+function handleAddServices(selectedServiceIds: string[]) {
+  const selectedServices =
+    props.biobank.services?.filter((service) =>
+      selectedServiceIds.includes(service.id)
+    ) ?? [];
+  const secvicesLabelValuePair = selectedServices.map((service) => {
```
Comment (chinook25) [inline apps/directory/src/components/biobankcards-components/BiobankCard.vue:122]: typo

File: apps/directory/src/components/biobankcards-components/BiobankCard.vue
Comment (chinook25) [inline apps/directory/src/components/biobankcards-components/BiobankCard.vue:1]: Group functions and computeds a bit more

File: apps/directory/src/components/biobankcards-components/CollectionsSection.vue [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          :to="'/collection/' + collectionDetail.id"
+          title="Collection details"
+          class="text-dark"
+        >
+          <h5>
+            <!-- do we need a icon, is its meaning clear arross types ? -->
+            <!-- <span
+              class="fa fa-server collection-icon fa-lg mr-2 text-primary"
+              aria-hidden="true"
+            /> -->
```
Comment (chinook25) [inline apps/directory/src/components/biobankcards-components/CollectionsSection.vue:41]: remove comment?

File: apps/directory/src/components/cart/NegotiatorSelection.vue [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    ) {
+      commercialAvailableCollections.value =
+        collectionsResponse.Collections.map((collection) => collection.id);
+      loading.value = false;
+    }
+  });
+
+const emit = defineEmits(["update:modelValue"]);
+
+const errorMessage = ref(""); // New reactive error state
```
Comment (chinook25) [inline apps/directory/src/components/cart/NegotiatorSelection.vue:186]: remove comment

File: apps/directory/src/components/cart/NegotiatorSelection.vue [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+  );
+}
+
+function removeAll() {
+  checkoutStore.removeAllFromSelection(props.bookmark);
+  emit("update:modelValue", false);
+}
+
+async function sendRequest() {
+  errorMessage.value = ""; // Reset error message before request
```
Comment (chinook25) [inline apps/directory/src/components/cart/NegotiatorSelection.vue:214]: remove comment

File: apps/directory/src/components/cart/NegotiatorSelection.vue [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+const modalFooterText = computed(
+  () => `${selectionCount.value} item(s) selected`
+);
+
+const selectedCollections = computed(() => checkoutStore.selectedCollections);
+
+const selectedServices = computed(() => checkoutStore.selectedServices);
+
+const nSelectedNonCommercialCollections = computed(() => {
+  let allCollectionIds: string[] = [];
```
Comment (chinook25) [inline apps/directory/src/components/cart/NegotiatorSelection.vue:240]: const

File: apps/directory/src/functions/bookmarkMapper.ts [hunk 6]
```diff
@@ -1,52 +1,45 @@
...
-
-  router.push({
-    name: router.currentRoute.name,
-    query: bookmark,
-  });
-}
-
-export async function applyBookmark(watchedQuery) {
+export async function applyBookmark(watchedQuery: Record<string, string>) {
+  console.log("applying bookmark", watchedQuery);
```
Comment (chinook25) [inline apps/directory/src/functions/bookmarkMapper.ts:9]: remove debug line

File: apps/directory/src/functions/bookmarkMapper.ts [hunk 7]
```diff
@@ -1,52 +1,45 @@
...
   const filtersStore = useFiltersStore();
-  let query = watchedQuery;
-  if (!query) {
-    const route = router.currentRoute.value;
-    if (!route.query) return;
-    query = route.query;
+
+  if (!watchedQuery || !(Object.keys(watchedQuery).length > 0)) {
+    return;
   }
```
Comment (chinook25) [inline apps/directory/src/functions/bookmarkMapper.ts:20]: guards should be at the top of functions

File: apps/directory/src/stores/checkoutStore.ts [hunk 8]
```diff
@@ -59,9 +97,64 @@ export const useCheckoutStore = defineStore("checkoutStore", () => {
...
+
+      setSearchHistory(
+        `Selected ${services
+          .map((nc) => nc.label)
+          .join(", ")} from ${biobankIdentifier}`
+      );
+
+      if (bookmark) {
+        checkoutValid.value = true;
+        // todo need to add service stuff to the bookmark
```
Comment (chinook25) [inline apps/directory/src/stores/checkoutStore.ts:136]: remove

File: apps/directory/src/stores/checkoutStore.ts [hunk 9]
```diff
@@ -213,11 +366,31 @@ export const useCheckoutStore = defineStore("checkoutStore", () => {
...
-              id: biobank.id,
-              externalId: biobank.id,
-              name: biobank.label,
-            },
+            // todo: is this needed?
+            // organization: {
+            //   id: biobank.value,
+            //   externalId: biobank.id,
+            //   name: biobank.label,
+            // },
```
Comment (chinook25) [inline apps/directory/src/stores/checkoutStore.ts:374]: remove

Comment (chinook25) [reply apps/directory/src/stores/checkoutStore.ts:374]: Make issue for this

File: apps/directory/src/stores/filtersStore.ts [hunk 10]
```diff
@@ -297,32 +302,35 @@ export const useFiltersStore = defineStore("filtersStore", () => {
...
   }
 
-  function updateFilter(filterName: string, value: any, fromBookmark: any) {
-    bookmarkTriggeredFilter.value = fromBookmark;
+  function updateFilter(
+    filterName: string,
+    value?: IFilterOption[] | string | boolean,
+    fromBookmark?: boolean
+  ) {
+    console.log("updateFilter", unref(value));
```
Comment (chinook25) [inline apps/directory/src/stores/filtersStore.ts:317]: debug line

File: apps/directory/src/views/ServiceReport.vue [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <!-- <pre>{{ service }}</pre> -->
```
Comment (chinook25) [inline apps/directory/src/views/ServiceReport.vue:143]: remove?

File: apps/directory/vite.config.js [hunk 12]
```diff
@@ -6,7 +6,7 @@ import monacoEditorPlugin from "vite-plugin-monaco-editor";
 
 const HOST =
   process.env.MOLGENIS_APPS_HOST || "https://bbmri-emx2-test.molgeniscloud.org";
-const SCHEMA = process.env.MOLGENIS_APPS_SCHEMA || "ERIC";
+const SCHEMA = process.env.MOLGENIS_APPS_SCHEMA || "directory";
```
Comment (chinook25) [inline apps/directory/vite.config.js:9]: undo

**Review by chinook25 — APPROVED** [review]

**Pull request merged**